integration/test/metric/metric_value_query.go

```diff
@@ -0,0 +1,97 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric
+
+import (
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"log"
+	"time"
+)
+
+var metricValueFetchers = []MetricValueFetcher{
+	&CPUMetricValueFetcher{},
+}
+
+func GetMetricFetcher(metricName string) (MetricValueFetcher, error) {
+	for _, fetcher := range metricValueFetchers {
+		if fetcher.isApplicable(metricName) {
+			return fetcher, nil
+		}
+	}
+	err := fmt.Errorf("No metric fetcher for metricName %v", metricName)
+	log.Printf("%s", err)
+	return nil, err
+}
+
+type MetricValueFetcher interface {
+	Fetch(namespace string, metricName string, stat Statistics) ([]float64, error)
+	fetch(namespace string, metricSpecificDimensions []types.Dimension, metricName string, stat Statistics) ([]float64, error)
+	isApplicable(metricName string) bool
+	getMetricSpecificDimensions() []types.Dimension
+}
+
+type baseMetricValueFetcher struct{}
+
+func (f *baseMetricValueFetcher) fetch(namespace string, metricSpecificDimensions []types.Dimension, metricName string, stat Statistics) ([]float64, error) {
+	ec2InstanceId := test.GetInstanceId()
+	instanceIdDimension := types.Dimension{
+		Name:  aws.String("InstanceId"),
+		Value: aws.String(ec2InstanceId),
+	}
+	dimensions := append(metricSpecificDimensions, instanceIdDimension)
+	metricToFetch := types.Metric{
+		Namespace:  aws.String(namespace),
+		MetricName: aws.String(metricName),
+		Dimensions: dimensions,
+	}
+
+	metricQueryPeriod := int32(60)
+	metricQuery := types.MetricDataQuery{
+		MetricStat: &types.MetricStat{
+			Metric: &metricToFetch,
+			Period: &metricQueryPeriod,
+			Stat:   aws.String(string(stat)),
+		},
+		Id: aws.String(metricName),
+	}
+	metricDataQueries := make([]types.MetricDataQuery, 1)
+	metricDataQueries[0] = metricQuery
```

Comment:
Similarly, can simplify this so we don't `make` an array and then set a value in it, we can just define the slice literal, like the above example for the CPU dimensions comment.